src/Symfony/Component/Finder/Iterator/FilterIterator.php

```diff
@@ -18,9 +18,24 @@
  *
  * @author Alex Bogomazov
  */
-abstract class FilterIterator extends \FilterIterator
+abstract class FilterIterator extends \FilterIterator implements \RecursiveIterator
```

Comment:
The return value is not memoized in all other filters, but in filter where having a recursive inner iterator could make sense. The filter on the contents of a file for example doesn't memoize.